src/particle_filter/include/particle_filter/matrix.h

```diff
@@ -0,0 +1,69 @@
+#ifndef MATRIX_H
```

Comment:
If you use Eigen library all of this is done for you, and will not have bugs.
If you use the ros tf library transforms are also done for you.